src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -31,33 +29,70 @@ import {
 } from '@/resources/assets/externalAssetsQuery';
 import { ethereumUtils } from '@/utils';
 import { queryClient } from '@/react-query';
+import { userAssetsStore } from '@/state/assets/userAssets';
+
+function omitNativePriceWorklet(
+  asset: ExtendedAnimatedAssetWithColors | null
+): Omit<ExtendedAnimatedAssetWithColors, 'nativePrice' | 'price'> | null {
+  'worklet';
+  if (!asset) return null;
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { nativePrice, price, ...rest } = asset;
+  return rest;
+}
+
+function getInitialInputValues() {
+  const initialSelectedInputAsset = userAssetsStore.getState().userAssets.values().next().value;
+  const initialBalance = Number(initialSelectedInputAsset?.balance.amount) ?? 0;
+  const initialNiceIncrement = findNiceIncrement(initialBalance);
+  const initialDecimalPlaces = countDecimalPlaces(initialNiceIncrement);
+
+  const initialInputAmount = Number(
+    niceIncrementFormatter({
+      incrementDecimalPlaces: initialDecimalPlaces,
+      inputAssetBalance: initialBalance,
+      inputAssetUsdPrice: initialSelectedInputAsset?.price?.value ?? 0,
+      niceIncrement: initialNiceIncrement,
+      percentageToSwap: 0.5,
+      sliderXPosition: SLIDER_WIDTH / 2,
+    })
+  );
+  const initialInputNativeValue = addCommasToNumber((initialInputAmount * (initialSelectedInputAsset?.price?.value ?? 0)).toFixed(2));
+
+  return {
+    initialInputAmount,
+    initialInputNativeValue,
+  };
+}
 
 export function useSwapInputsController({
   focusedInput,
-  lastTypedInput,
   inputProgress,
-  outputProgress,
   internalSelectedInputAsset,
   internalSelectedOutputAsset,
   isFetching,
   isQuoteStale,
-  sliderXPosition,
+  lastTypedInput,
+  outputProgress,
   quote,
+  sliderXPosition,
 }: {
   focusedInput: SharedValue<inputKeys>;
-  lastTypedInput: SharedValue<inputKeys>;
   inputProgress: SharedValue<number>;
-  outputProgress: SharedValue<number>;
   internalSelectedInputAsset: SharedValue<ExtendedAnimatedAssetWithColors | null>;
   internalSelectedOutputAsset: SharedValue<ExtendedAnimatedAssetWithColors | null>;
   isFetching: SharedValue<boolean>;
   isQuoteStale: SharedValue<number>;
-  sliderXPosition: SharedValue<number>;
+  lastTypedInput: SharedValue<inputKeys>;
+  outputProgress: SharedValue<number>;
   quote: SharedValue<Quote | CrosschainQuote | QuoteError | null>;
+  sliderXPosition: SharedValue<number>;
 }) {
+  const { initialInputAmount, initialInputNativeValue } = getInitialInputValues();
+
   const inputValues = useSharedValue<{ [key in inputKeys]: number | string }>({
-    inputAmount: 0,
-    inputNativeValue: 0,
+    inputAmount: initialInputAmount,
```

Comment:
open question / note to self: what's happening for the scenario for internal deeplinks if we want to "BUY" a specific token

Reply:
Follow-up question before I can answer that one, but if we're prefilling an asset to buy, are we prefilling the asset to sell as the largest asset on the same network?

In not: We can do this no problem and just leave the output values as `0`
If so: We'll have to prefetch a quote to show the initial output values based on that quote.